### PR TITLE
update doc for IndexMerge GA

### PR DESCRIPTION
**Pull request by guo-shaoge**
Signed-off-by: guo-shaoge <shaoge1994@163.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@guo-shaoge Please involve a technical review and a PM review. Thanks!

File: system-variables.md [hunk 1]
```diff
@@ -610,7 +610,7 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 ### `tidb_enable_index_merge` <span class="version-mark">从 v4.0 版本开始引入</span>
 
 - 作用域：SESSION | GLOBAL
-- 默认值：`OFF`
+- 默认值：`ON`
```
Comment (TomShawn) [inline system-variables.md:613]:
```suggestion
- 默认值：`ON` 
```

File: explain-index-merge.md [hunk 2]
```diff
@@ -36,13 +36,11 @@ explain select * from t use index(idx_a, idx_b) where a > 1 or b > 1;
...
 > **注意：**
 >
-> 目前，TiDB 的 `IndexMerge` 特性在 TiDB 4.0.0-rc.1 版本中默认关闭。同时 4.0 版本中的 `IndexMerge` 目前支持的场景仅限于析取范式（`or` 连接的表达式），暂不支持合取范式（`and` 连接的表达式）。开启 `IndexMerge` 特性有以下方法：
->
-> - 设置系统变量 `tidb_enable_index_merge=1`；
->
-> - 在查询中使用 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-)。
->
-> SQL Hint 的优先级高于系统变量。
+> - TiDB 的 `IndexMerge` 特性在 v5.4.0 及之后的版本默认打开 (tidb_enable_index_merge)
```
Comment (XuHuaiyu) [inline explain-index-merge.md:39]: 1. tidb_enable_index_merge 上加个超链接到对应的系统变量说明上
2. 说清楚由低版本升级，和新集群部署的行为区别

Comment (guo-shaoge) [reply explain-index-merge.md]: 1. done
2. 放在 system-variables.md 中解释

**Review by XuHuaiyu — APPROVED** [review]
LGTM

File: system-variables.md [hunk 3]
```diff
@@ -610,9 +610,16 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- 默认值：`ON` 
 - 这个变量用于控制是否开启 index merge 功能。
 
+> **注意：**
+>
+> 1. 版本号小于 v4.0.0 的集群升级到版本号大于等于 v5.4.0 的集群，开关默认保持关闭。防止升级后计划发生变化导致回退。
+> 2. 版本号大于等于 v4.0.0 的集群升级到版本号大于等于 v5.4.0 的集群，开关保持升级前的状态。
+> 3. 版本号大于等于 v5.4.0 的新建集群，开关默认保持开启。
+
+
```
Comment (shichun-0415) [inline system-variables.md:622]:
```suggestion
> - 当集群从 v4.0.0 以下版本升级到 v5.4.0 及以上版本时，该变量开关默认关闭，防止升级后计划发生变化导致回退。
> - 当集群从 v4.0.0 及以上版本升级到 v5.4.0 及以上版本时，该变量开关保持升级前的状态。
> - 对于 v5.4.0 及以上版本的新建集群，该变量开关默认开启。

```
The modification is as follows:
1. Optimized description about the version information
2. Changed the ordered list into an unordered list.
3. Deleted a redundant blank line.

File: explain-index-merge.md [hunk 4]
```diff
@@ -36,13 +36,11 @@ explain select * from t use index(idx_a, idx_b) where a > 1 or b > 1;
...
 > **注意：**
 >
-> 目前，TiDB 的 `IndexMerge` 特性在 TiDB 4.0.0-rc.1 版本中默认关闭。同时 4.0 版本中的 `IndexMerge` 目前支持的场景仅限于析取范式（`or` 连接的表达式），暂不支持合取范式（`and` 连接的表达式）。开启 `IndexMerge` 特性有以下方法：
->
-> - 设置系统变量 `tidb_enable_index_merge=1`；
->
-> - 在查询中使用 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-)。
->
-> SQL Hint 的优先级高于系统变量。
+> - TiDB 的 `IndexMerge` 特性在 v5.4.0 及之后的版本默认打开 ([tidb_enable_index_merge](/system-variables.md#tidb_enable_index_merge-从-v4.0-版本开始引入))
```
Comment (moyun) [inline explain-index-merge.md:39]: 这个词“IndexMerge”是怎么写好？ 要不要写成 “Index Merge” 跟MySQL，Oracle接轨一下, 并且下文中也出现了 `index merge`的写法。似应统一为好？
另外要不要给一个中文的翻译词汇？还是就这么叫了...

Comment (moyun) [reply explain-index-merge.md:39]: 我觉得叫“索引合并优化”听上去比较顺

Comment (shichun-0415) [reply explain-index-merge.md:39]: > 我觉得叫“索引合并优化”听上去比较顺

@moyun @guo-shaoge “索引合并优化”对应的是 MySQL 的 “Index Merge Optimization”，所以咱们这个地方叫“索引合并”就可以了吧。写法可以跟 MySQL 保持一致，用 Index Merge。

File: explain-index-merge.md [hunk 5]
```diff
@@ -1,22 +1,21 @@
...
 
-# 用 EXPLAIN 查看开启 IndexMerge 的 SQL 执行计划
+# 用 EXPLAIN 查看索引合并的 SQL 执行计划
 
-`IndexMerge` 是 TiDB v4.0 中引入的一种对表的新访问方式。在这种访问方式下，TiDB 优化器可以选择对一张表使用多个索引，并将每个索引的返回结果进行合并。在某些场景下，这种访问方式能够减少大量不必要的数据扫描，提升查询的执行效率。
+索引合并是 TiDB v4.0 中引入的一种对表的新访问方式。在这种访问方式下，TiDB 优化器可以选择对一张表使用多个索引，并将每个索引的返回结果进行合并。在某些场景下，这种访问方式能够减少大量不必要的数据扫描，提升查询的执行效率。
 
 ```sql
-EXPLAIN SELECT * from t where a = 1 or b = 1;
+explain select * from t where a = 1 or b = 1;
```
Comment (shichun-0415) [inline explain-index-merge.md:11]:
```suggestion
EXPLAIN SELECT * FROM t WHERE a = 1 or b = 1;
```

File: explain-index-merge.md [hunk 6]
```diff
@@ -28,22 +27,18 @@ explain select * from t use index(idx_a, idx_b) where a > 1 or b > 1;
...
-> - 设置系统变量 `tidb_enable_index_merge=1`；
->
-> - 在查询中使用 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-)。
->
-> SQL Hint 的优先级高于系统变量。
-
+> - TiDB 的索引合并特性在 v5.4.0 及之后的版本默认打开，即 [tidb_enable_index_merge](/system-variables.md#tidb_enable_index_merge-从-v4.0-版本开始引入)) 设置为 `ON`
+> - 如果查询中使用了 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-) ，无论开关是否开启，都会强制使用该特性。当过滤条件中有无法下推的表达式时，必须使用 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-) 才能开启索引合并
+> - 索引合并目前仅支持析取范式（`or` 连接的表达式），不支持合取范式（`and` 连接的表达式）
+> - 索引合并目前无法在临时表上使用
```
Comment (shichun-0415) [inline explain-index-merge.md:41]:
```suggestion
> - TiDB 的索引合并特性在 v5.4.0 及之后的版本默认打开，即 [tidb_enable_index_merge](/system-variables.md#tidb_enable_index_merge-从-v4.0-版本开始引入)) 设置为 `ON`。
> - 如果查询中使用了 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-) ，无论开关是否开启，都会强制使用索引合并特性。当过滤条件中有无法下推的表达式时，必须使用 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-) 才能开启索引合并。
> - 索引合并目前仅支持析取范式（`or` 连接的表达式），不支持合取范式（`and` 连接的表达式）。
> - 索引合并目前无法在临时表上使用。
```

**Review by shichun-0415 — APPROVED** [review]
rest LGTM

File: explain-index-merge.md (hunk 5 above)
Comment (guo-shaoge) [reply explain-index-merge.md]: done

File: explain-index-merge.md [hunk 7]
```diff
@@ -28,21 +27,18 @@ explain select * from t use index(idx_a, idx_b) where a > 1 or b > 1;
...
 > **注意：**
 >
-> 目前，TiDB 的 `IndexMerge` 特性在 TiDB 4.0.0-rc.1 版本中默认关闭。同时 4.0 版本中的 `IndexMerge` 目前支持的场景仅限于析取范式（`or` 连接的表达式），暂不支持合取范式（`and` 连接的表达式）。开启 `IndexMerge` 特性有以下方法：
->
-> - 设置系统变量 `tidb_enable_index_merge=1`；
->
-> - 在查询中使用 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-)。
->
-> SQL Hint 的优先级高于系统变量。
+> - TiDB 的索引合并特性在 v5.4.0 及之后的版本默认打开，即 [tidb_enable_index_merge](/system-variables.md#tidb_enable_index_merge-从-v4.0-版本开始引入)) 设置为 `ON`。
```
Comment (qiancai) [inline explain-index-merge.md:38]:
```suggestion
> - TiDB 的索引合并特性在 v5.4.0 及之后的版本默认开启，即 [`tidb_enable_index_merge`](/system-variables.md#tidb_enable_index_merge-从-v4.0-版本开始引入) 为 `ON`。
```

File: explain-index-merge.md [hunk 8]
```diff
@@ -28,21 +27,18 @@ explain select * from t use index(idx_a, idx_b) where a > 1 or b > 1;
...
+> - 如果查询中使用了 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-) ，无论开关是否开启，都会强制使用索引合并特性。当过滤条件中有无法下推的表达式时，必须使用 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-) 才能开启索引合并。
```
Comment (qiancai) [inline explain-index-merge.md:39]:
```suggestion
> - 如果查询中使用了 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-) ，无论 `tidb_enable_index_merge` 开关是否开启，都会强制使用索引合并特性。当过滤条件中有无法下推的表达式时，必须使用 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-) 才能开启索引合并。
```

File: explain-index-merge.md [hunk 9]
```diff
@@ -1,23 +1,22 @@
...
-title: 用 EXPLAIN 查看开启 IndexMerge 的 SQL 执行计划
+title: 用 EXPLAIN 查看索引合并的 SQL 执行计划
 summary: 了解 TiDB 中 EXPLAIN 语句返回的执行计划信息。
 ---
 
-# 用 EXPLAIN 查看开启 IndexMerge 的 SQL 执行计划
+# 用 EXPLAIN 查看索引合并的 SQL 执行计划
 
-`IndexMerge` 是 TiDB v4.0 中引入的一种对表的新访问方式。在这种访问方式下，TiDB 优化器可以选择对一张表使用多个索引，并将每个索引的返回结果进行合并。在某些场景下，这种访问方式能够减少大量不必要的数据扫描，提升查询的执行效率。
+索引合并是 TiDB v4.0 中引入的一种对表的新访问方式。在这种访问方式下，TiDB 优化器可以选择对一张表使用多个索引，并将每个索引的返回结果进行合并。在某些场景下，这种访问方式能够减少大量不必要的数据扫描，提升查询的执行效率。
```
Comment (qiancai) [inline explain-index-merge.md:8]:
```suggestion
索引合并是从 TiDB v4.0 起引入的一种对表的新访问方式。在这种访问方式下，TiDB 优化器可以选择对一张表使用多个索引，并将每个索引的返回结果进行合并。在某些场景下，这种访问方式能够减少大量不必要的数据扫描，提升查询的执行效率。
```

File: explain-index-merge.md [hunk 10]
```diff
@@ -28,21 +27,18 @@ explain select * from t use index(idx_a, idx_b) where a > 1 or b > 1;
...
 > **注意：**
 >
-> 目前，TiDB 的 `IndexMerge` 特性在 TiDB 4.0.0-rc.1 版本中默认关闭。同时 4.0 版本中的 `IndexMerge` 目前支持的场景仅限于析取范式（`or` 连接的表达式），暂不支持合取范式（`and` 连接的表达式）。开启 `IndexMerge` 特性有以下方法：
->
-> - 设置系统变量 `tidb_enable_index_merge=1`；
->
-> - 在查询中使用 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-)。
->
-> SQL Hint 的优先级高于系统变量。
+> - TiDB 的索引合并特性在 v5.4.0 及之后的版本默认开启，即 [`tidb_enable_index_merge`](/system-variables.md#tidb_enable_index_merge-从-v4.0-版本开始引入) 为 `ON`。
```
Comment (shichun-0415) [inline explain-index-merge.md:38]:
```suggestion
> - TiDB 的索引合并特性在 v5.4.0 及之后的版本默认开启，即 [`tidb_enable_index_merge`](/system-variables.md#tidb_enable_index_merge-从-v40-版本开始引入) 为 `ON`。
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**